<?xml version="1.0"?>
<launch>
  <arg name="publish_cmd" default="true" />

  <arg name="base_global_planner" default="global_planner/GlobalPlanner"/>
  <arg name="base_local_planner" default="base_local_planner/TrajectoryPlannerROS"/>

  <node pkg="move_base" type="move_base" respawn="false" name="move_base" output="screen">

    <param name="base_global_planner" value="$(arg base_global_planner)" />
    <param name="base_local_planner" value="$(arg base_local_planner)"/>

    <rosparam file="$(find racecar_navigation)/config/planner.yaml" command="load"/>
    <param name="SBPLLatticePlanner/primitive_filename" value="$(find racecar_navigation)/config/car_.mprim" />

    <!-- external controller -->
    <remap from="/cmd_vel" to="/fake_cmd" unless="$(arg publish_cmd)" />
    <remap from="/move_base/SBPLLatticePlanner/plan" to="/move_base/global_plan"/>
    <remap from="/move_base/GlobalPlanner/plan" to="/move_base/global_plan"/>

    <!-- observation sources located in costmap_common.yaml -->
    <rosparam file="$(find racecar_navigation)/config/costmap_common.yaml" command="load" ns="global_costmap" />
    <rosparam file="$(find racecar_navigation)/config/costmap_common.yaml" command="load" ns="local_costmap" />

    <!-- local costmap, needs size -->
    <rosparam file="$(find racecar_navigation)/config/costmap_local.yaml" command="load" ns="local_costmap" />
    <param name="local_costmap/width" value="10.0"/>
    <param name="local_costmap/height" value="10.0"/>

    <!-- static global costmap, static map provides size -->
    <rosparam file="$(find racecar_navigation)/config/costmap_global.yaml" command="load" ns="global_costmap" />
  </node>

</launch>